test/functional/fixtures/page-js-errors/test.js

```diff
@@ -56,11 +66,143 @@ describe('Should ignore an js-error on the page if the skipJsErrors option is se
         return runTests('./testcafe-fixtures/error-after-click-test.js', 'Click button', { skipJsErrors: true });
     });
 
-    it ('unhandled Promise rejection', () => {
+    it('unhandled Promise rejection', () => {
         return runTests('./testcafe-fixtures/unhandled-promise-rejection-test.js', 'Click button',
             {
                 skipJsErrors: true,
                 only:         'chrome',
             });
     });
 });
+
+const expectFailAttempt = (errors, expectedMessage) => {
+    Object.values(errors).forEach(err => {
+        const error = castArray(err);
+
+        expect(error[0].message || error[0]).contains(expectedMessage);
+    });
+};
+
+// TODO: fix tests for Debug task
+(experimentalDebug ? describe.skip : describe)('Customize SkipJSErrors (GH-2775)', () => {
+    describe('TestController method', () => {
+        it('Should skip JS errors without param', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors without param');
+        });
+
+        it('Should skip JS errors with boolean param', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors with boolean param');
+        });
+
+        it('Should skip JS errors with message and stack options', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors with multiple options');
+        });
+
+        it('Should skip JS errors with callback function', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors with callback function');
+        });
+
+        it('Should skip JS errors with async callback function', async () => {
```

Comment:
This name can be confusing since we don't support async functions for `ClientFunction`.
Let's rename to `Should skip JS errors with callback function returning Promise`.